packages/world-modules/src/modules/erc20-puppet/ERC20Module.sol

```diff
@@ -83,6 +88,12 @@ contract ERC20ModuleRegistrationLibrary {
    * Register systems and tables for a new ERC20 token in a given namespace
    */
   function register(IBaseWorld world, bytes14 namespace) public {
+    // Register the namespace if it doesn't exist yet
+    ResourceId tokenNamespace = WorldResourceIdLib.encodeNamespace(namespace);
+    if (!ResourceIds.getExists(tokenNamespace)) {
```

Comment:
(not blocking)

since this is a common check, I wonder if we should add some helpers/abstractions for cases this like
```solidity
function namespaceExists(ResourceId resourceId)
function namespaceExists(bytes14 namespace)
```

mostly thinking about it feeling a little weird to sprinkle `ResourceIds` table usage everywhere, but maybe we kinda have to due to gas

Reply:
agree this might be useful, would not do it in this PR though to keep the scope smaller